packages/rancher-monitoring/rancher-monitoring.patch

```diff
@@ -1843,7 +1834,7 @@ diff -x '*.tgz' -x '*.lock' -uNr packages/rancher-monitoring/charts-original/tem
 +apiVersion: rbac.authorization.k8s.io/v1
 +kind: ClusterRole
```

Comment:
This should be a Role to scope this properly. As a clusterRole it would grant perms to ALL resources listed in the cluster if someone used a clusterRoleBinding with it.

Reply:
@dramich based on the fact that we also want to remove ConfigMap / Secret permissions from the `monitoring-<admin/view/edit>` cluster roles based on https://github.com/rancher/rancher/issues/29013, I refactored the logic for the Roles to instead introduce two different sets of roles:

`monitoring-<admin/view/edit>`: only allows access to Prometheus CRs; previously, these permissions also allowed users to modify ConfigMaps / Secrets, but this has since been removed. I've also documented in the `CHANGELOG.md` that the expectation is for cluster admins to bind this role to users using a `ClusterRoleBinding` in order to allow users full access to view / edit Prometheus CRs across the entire cluster.

`monitoring-config-<admin/view/edit>`: this only allows access to ConfigMaps and Secrets. I've also documented in the `CHANGELOG.md` that the expectation is for cluster admins to bind this role to users using a `RoleBinding` in order to allow users limited access to view / edit ConfigMaps / Secrets only in specific namespaces.

Both sets of roles will be aggregated up to the expected default Kubernetes Roles if that option is provided.

Reply:
I don't agree with creating the 2nd sets of roles as ClusterRoles. It feels like we are handing out a foot gun for a role called monitoring-config-edit but in reality it gives you access to secrets. Docs are a suggestion, not what people are really going to do in the wild once this is deployed. 
The first set need to be ClusterRoles in order to roll up to their k8s parents. But the 2nd set does not need to role up as those k8s default roles already have access to secrets/configmaps. If you want to add those 2nd roles I think they should be Roles not ClusterRoles.

Reply:
Hmm I see your point. If we want to make them `Roles`, I think it will need to be split into `monitoring-config-<admin|edit|view>` which is a set of `Roles` in the `cattle-monitoring-system` namespace and `monitoring-dashboard-<admin|edit|view>`, which is a set of `Roles` in the `cattle-dashboards` namespace. Aggregation will also be removed, as you mentioned.

Will make the appropriate changes in the next commit.